src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -92,6 +92,116 @@ public EntityAIStructureMiner(@NotNull final JobMiner job)
         worker.setCanPickUpLoot(true);
     }
 
+    private static boolean isOre(final Block block)
+    {
+        //TODO make this more sophisticated
+        return block instanceof BlockOre;
+    }
+
+    private static void setNodeStatusForDirection(@NotNull final Node node, final int direction, final Node.NodeStatus status)
+    {
+        if (direction == 1)
+        {
+            node.setDirectionPosX(status);
+        }
+        else if (direction == 2)
+        {
+            node.setDirectionNegX(status);
+        }
+        else if (direction == 3)
+        {
+            node.setDirectionPosZ(status);
+        }
+        else if (direction == 4)
+        {
+            node.setDirectionNegZ(status);
+        }
+    }
+
+    private static Node.NodeStatus getNodeStatusForDirection(@NotNull final Node node, final int direction)
+    {
+        if (direction == 1)
+        {
+            return node.getDirectionPosX();
+        }
+        else if (direction == 2)
+        {
+            return node.getDirectionNegX();
+        }
+        else if (direction == 3)
+        {
+            return node.getDirectionPosZ();
+        }
+        else if (direction == 4)
+        {
+            return node.getDirectionNegZ();
+        }
+        //Cannot happen, so send something that blocks mining
+        return Node.NodeStatus.LADDER;
+    }
+
+    private static int invertDirection(final int direction)
+    {
+        if (direction == 1)
+        {
+            return 2;
+        }
+        else if (direction == 2)
+        {
+            return 1;
+        }
+        else if (direction == 3)
+        {
+            return 4;
+        }
+        else if (direction == 4)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)